drivers/volume/kdmp/kdmp.go

```diff
@@ -550,11 +549,20 @@ func (k *kdmp) StartRestore(
 
 func (k *kdmp) CancelRestore(restore *storkapi.ApplicationRestore) error {
 	for _, vInfo := range restore.Status.Volumes {
-		crName := getGenericCRName(restore.Name, vInfo.SourceNamespace, vInfo.PersistentVolumeClaim)
-		if err := kdmpShedOps.Instance().DeleteDataExport(crName, vInfo.SourceNamespace); err != nil {
+		val, ok := restore.Spec.NamespaceMapping[vInfo.SourceNamespace]
+		if !ok {
+			return fmt.Errorf("restore namespace mapping not found: %s", vInfo.SourceNamespace)
+
+		}
+		restoreNamespace := val
+		crName := getGenericCRName(restore.Name, restoreNamespace, vInfo.PersistentVolumeClaim)
+		if err := kdmpShedOps.Instance().DeleteDataExport(crName, restoreNamespace); err != nil {
 			logrus.Errorf("failed to delete data export CR: %v", err)
 			return err
 		}
+		if err := kdmpShedOps.Instance().DeleteVolumeBackup(prefixRestore+"-"+crName, restoreNamespace); err != nil {
+			logrus.Tracef("failed to delete volume backup CR:%s/%s, err: %v", restoreNamespace, crName, err)
```

Comment:
Don't want to return failure from here?

Reply:
yes, since this is will be final stage where backup/restore is already completed. If we return err, we get immediate reconciler event